### PR TITLE
Origin/chef provisioner reporting

### LINKED ISSUE #4464: Chef provisioner: add enable_reporting flag to client.rb
Reporting is premium feature of Chef Server 12. And is working fine for free for 25 nodes, after that you will get errors from Chef Server. It's not major issue, but it's a small change.

Default value(if skipped in config): enable_reporting       true

We need to make it false, if required.
https://docs.chef.io/reporting.html

### PR DESCRIPTION
**Pull request by partamonov**
Fix #4464. Add enable_report flag for chef provisioner

### CODE REVIEW

File: builtin/provisioners/chef/resource_provisioner.go [hunk 1]
```diff
@@ -84,6 +86,7 @@ type Provisioner struct {
 	ServerURL            string      `mapstructure:"server_url"`
 	SkipInstall          bool        `mapstructure:"skip_install"`
 	SSLVerifyMode        string      `mapstructure:"ssl_verify_mode"`
+	EnableReporting      string      `mapstructure:"enable_reporting"`
```
Comment (svanharmelen) [inline builtin/provisioners/chef/resource_provisioner.go]: Wouldn't it make more sense to define this like:

```go
DisableReporting bool `mapstructure:"disable_reporting"`
```

This way it's really clear that by default this is on (which is the default for Chef as well), and at the same time we can make better use of Go's type safety.

Of course the docs should then also be updated just a little and the template should be changed to something like:

```go
{{ if .DisableReporting }}enable_reporting false{{ end }}
```

**Comment by svanharmelen** [comment]
@partamonov nice addition! Just a small comment about the chosen naming/logic but nothing major :wink:

**Comment by partamonov** [comment]
I decided it's better to have same name as in chef docs, easy to find and reference.

**Comment by partamonov** [comment]
@svanharmelen thank you for review and comments.

**Comment by svanharmelen** [comment]
LGMT! Thanks!

**Pull request merged**